src/libraries/System.Private.CoreLib/src/System/Index.cs

```diff
@@ -15,7 +15,7 @@ namespace System
     /// int lastElement = someArray[^1]; // lastElement = 5
     /// </code>
     /// </remarks>
-#if SYSTEM_PRIVATE_CORELIB
+#if SYSTEM_PRIVATE_CORELIB || MICROSOFT_BCL_MEMORY
```

Comment:
Do we have other projects today that are building in Index/Range as internal because they're downlevel? If yes, can those switch to referencing this package instead?

Reply:
I found https://github.com/dotnet/runtime/blob/33ca32d09043f3acb646093b4f653ddcfade5adc/src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.csproj#L29 is the only place using it. I can try to remove it there and let the project depend on the new package. It is ns2.0 project. 

CC @jkoritzinsky for any feedback.

Reply:
This is done too. I made ComInterfaceGenerator reference the `Microsoft.Bcl.Memory` and removed including the Index code in the project. I did run the tests successfully too.

Reply:
You need to ensure that we deploy the Microsoft.Bcl.Memory netstandard2.0 asset next to the source generators in Microsoft.NETCore.App.Ref, otherwise you'll break the generators. Other than that, I'm fine with having it depend on the new package.